WordPress/gutenberg
# Introduce section container selection when assembling patterns (zoom out mode)

## What?


This PR introduces the notion of a section root block, much like we have the grouping block. This is the root block of all sections. By default this root is a group block with the tag `main`, but in certain contexts editors would want to set it to something else. For example the site editor while editing a page will set the section root block to be the post content block since that's where the sections are.

In Zoomed out mode:

- disable all blocks that are not "the main content" (a new selector is added for this) - e.g. header, footer...etc
- allow basic interactions on top level children within the "main content"
- disable interacting with low level blocks

## Testing Instructions

1. Enable the zoom out mode Gutenberg experiment. 
2. Open the site editor.
3. Ensure you have patterns inserted on a page or template.
4. Engage zoom out view from the editor header.
5. See that you can only select top-level blocks/patterns.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/444434/744c3ebb-5c4a-4562-99fd-27ddbc8abe42
